src/node/miner.h

```diff
@@ -172,6 +167,8 @@ class BlockAssembler
     inline static std::optional<int64_t> m_last_block_weight{};
 
 private:
+    Options m_options;
```

Comment:
could make that `const` by also adding:

```c++
static BlockAssembler::Options clamp_options(const BlockAssembler::Options& opt)
{
    Options clamped{opt};
    clamped.nBlockMaxWeight = std::clamp<size_t>(clamped.nBlockMaxWeight, 4000, DEFAULT_BLOCK_MAX_WEIGHT);
    return clamped;
}
```

and setting `m_options{clamped(options)}` in the constructor.